lib/models/device.coffee

```diff
@@ -88,9 +92,33 @@ getDeviceModel = (deps, opts) ->
 		if semver.lt(version, minVersion)
 			throw new Error("Incompatible supervisor version: #{version} - must be >= #{minVersion}")
 
-	addApplicationName = (device) ->
+	###*
+	# @summary Get Dashboard URL for a specific device
+	# @function getDashboardUrl
+	#
+	# @param {Object} options - options
+	# @param {Number} options.appId - Application id
+	# @param {Number} options.deviceId - Device id
+	#
+	# @returns {String} - Dashboard URL for the specific device
+	# @throws Exception if either appId or deviceId are empty
+	#
+	# @example
+	# dashboardDeviceUrl = resin.models.device.getDashboardUrl({ appId: 123, deviceId: 456 })
+	###
+	exports.getDashboardUrl = getDashboardUrl = (options = {}) ->
+		isFinite = require('lodash/isFinite')
+		url = require('url')
```

Comment:
@pimterry makes sense, I've bubbled the `require`'s up since this is how it's done currently